drivers/adc/ad7799/iio_ad7799.h

```diff
@@ -0,0 +1,148 @@
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#ifndef IIO_AD7799_H
+#define IIO_AD7799_H
+
+#include "iio_types.h"
+
+/** AD7799 IIO register read */
+ssize_t ad7799_iio_reg_read(void *device, char *buf, size_t len,
+			    const struct iio_ch_info *channel, intptr_t priv);
+/** AD7799 IIO Channel read */
+ssize_t ad7799_iio_channel_read(void *device, char *buf, size_t len,
+				const struct iio_ch_info *channel, intptr_t priv);
+/** AD7799 IIO Gain write read */
+ssize_t ad7799_iio_gain_read(void *device, char *buf, size_t len,
+			     const struct iio_ch_info *channel, intptr_t priv);
+/** AD7799 IIO Gain write */
+ssize_t ad7799_iio_gain_write(void *device, char *buf, size_t len,
+			      const struct iio_ch_info *channel, intptr_t priv);
+
+/** ADC IIO channel attributes*/
+struct iio_attribute ad7799_iio_channel_attributes[] = {
+	{
+		.name = "volts",
+		.show = ad7799_iio_channel_read,
+		.store = NULL
+	},
+	END_ATTRIBUTES_ARRAY,
+};
+
+/** Channel Scan Type */
+struct scan_type channel_scan_type = {
+	.sign = 'u',
+	.realbits = 12,
+	.storagebits = 24,
+	.shift = 0,
+	.is_big_endian = false
+};
+
+/** IIO ADC Channels */
+struct iio_channel ad7799_iio_channels[] = {
+	{
```

Comment:
nitpick: if you make a define for channel declaration you can avoid a lot of code duplication.

Reply:
Sure, I can do :)

Reply:
I think Andrei was meaning something like this:
![image](https://user-images.githubusercontent.com/16192750/103923390-b9521d80-511d-11eb-827c-74b395102e39.png)

Reply:
The old file was overwritten locally when I applied the astyle.
should be fine now.